### DISCUSSION TITLE
About Python codegen: Do we have to write one query per edgeql file?

### DISCUSSION THREAD

**Discussion by jenish2014** [geldata/gel#4742] [Q&A]
One query per edgeql file -> codegen -> one output file

Can we put all our queries in one single file delimited by ';' or some other separator and feed to codegen?

**Comment by fantix** [comment]
> Can we put all our queries in one single file delimited by ';'

Actually you can do that; it's just that the result is not what you expected. An `.edgeql` file with multiple semicolon-separated statements is considered a "script" that is always executed as an atomic operation, and only the result of the last statement is returned.

What is your use case that multiple `.edgeql` files are not acceptable/ideal?

**Reply by arogozhnikov** [reply]
I think isolated .edgeql are a big disadvantage. 

As author noted, there are many files (worse - interleaving), and they completely clutter the navigation sidebar - meaning, if you want to find some other file, you first need to close the folder to use navigator again (yes, I know about search over filenames, but navigator side panel exists in all IDEs for reasons). 

If you use auto-point to current file in navigator, you have to repeatedly close folder in navigator.

Second part is grouping of queries into logical blocks - keeping adjacent code together. 
This helps with reading and editing - you immediately see other places that may need changes, and can compare code without opening multiple files in separate panes.

I wonder if there is any argument to keep them in separate files?

**Comment by jenish2014** [comment]
Not a use case, its just that I have directories with 50-60 edgeql files and they have merely 1-2 lines of query written in them. I am making simple Blog website and its unwieldy to have so many files then sifting through them to find a query i want to edit.

It would get out of control for big project like e-commerce or similar. It would be nice to have one file with all the related queries like - 
PostQueries.edgeql 
Account.edgeql
Comments.edgeql ...
